Python/fileutils.c

```diff
@@ -1512,7 +1512,9 @@ set_inheritable(int fd, int inheritable, int raise, int *atomic_flag_works)
             request = FIOCLEX;
         err = ioctl(fd, request, NULL);
         if (!err) {
-            ioctl_works = 1;
+            if (_Py_atomic_load_int_relaxed(&ioctl_works) == -1) {
```

Comment:
Should we use the atomic operations on non free-threads build?

Reply:
Yes, it's fine. There is no performance implication -- these relaxed load/stores get compiled to the same instructions as non-atomic operations.

Reply:
Got it, thanks for the explanation!